### PR TITLE
feat: Add 'unsafe-to-chain-command' rule

### PR DESCRIPTION
**Pull request by Erik-Outreach**

### CODE REVIEW

File: lib/rules/unsafe-to-chain-command.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    node = node.callee.object
+  }
+
+  return false
+}
+
+function isActionUnsafeToChain (node) {
+  // commands listed in the documentation with text: 'It is unsafe to chain further commands that rely on the subject after xxx'
+  const unsafeToChainActions = ['blur', 'clear', 'click', 'check', 'dblclick', 'each', 'focus', 'rightclick', 'screenshot', 'scrollIntoView', 'scrollTo', 'select', 'selectFile', 'spread', 'submit', 'type', 'trigger', 'uncheck', 'wait', 'within']
```
Comment (BlueWinds) [inline lib/rules/unsafe-to-chain-command.js:44]: Minor note: In Cypress 12.0.0 - 12.2.0, `should` and `and` are also unsafe. This is a bug we'll be fixing ASAP, likely in 12.3.0 - see https://github.com/cypress-io/cypress/issues/25134 for discussion and details.

**Review by BlueWinds — APPROVED** [review]

**Comment by Erik-Outreach** [comment]
@BlueWinds tks for the approval! I'm aware you're working on merging this to the core repo, but any chance we get this merged and released in this package?

**Comment by BlueWinds** [comment]
@Erik-Outreach - Yeah, for sure. Was just reaching out to the rest of the team to ask about the release process for this repo, and to get a couple more pairs of eyes on it.

File: README.md [hunk 2]
```diff
@@ -121,6 +121,7 @@ Rules with a check mark (✅) are enabled by default while using the `plugin:cyp
 | ✅  | [no-assigning-return-values](./docs/rules/no-assigning-return-values.md)   | Prevent assigning return values of cy calls                     |
 | ✅  | [no-unnecessary-waiting](./docs/rules/no-unnecessary-waiting.md)           | Prevent waiting for arbitrary time periods                      |
 | ✅  | [no-async-tests](./docs/rules/no-async-tests.md)                           | Prevent using async/await in Cypress test case                  |
+| ✅  | [unsafe-to-chain-command](./docs/rules/unsafe-to-chain-command.md)         | Prevent chaining from unsafe to chain commands                  |
```
Comment (chrisbreiding) [inline README.md:124]: This wasn't added to the [recommended rules](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/lib/config/recommended.js), so it shouldn't be marked as enabled by default.

Comment (chrisbreiding) [reply README.md:124]: Or should be added to the recommended rules.

Comment (Erik-Outreach) [reply README.md:124]: Added as a new PR: https://github.com/cypress-io/eslint-plugin-cypress/pull/119

File: lib/rules/unsafe-to-chain-command.js (hunk 1 above)
Comment (Erik-Outreach) [reply lib/rules/unsafe-to-chain-command.js:44]: Or maybe wait command should be considered safe to chain (documentation was updated after this PR?). If that's the way to go, we just need to merge this PR to get it removed: https://github.com/cypress-io/eslint-plugin-cypress/pull/128

Comment (websitevirtuoso) [reply lib/rules/unsafe-to-chain-command.js:44]: Yes. very confusing I am using this a lot

      cy.getBySel('post.title').find('input').clear().type(newPost.title)
      but not my ESLINT started to show warning

**Pull request merged**